src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,802 @@
+            if storage_space_available:
+                outbound_connectivity_check_path = os.path.join(filepath_with_timestamp, "Outbound_Network_Connectivity_Check.txt")
+                with open(outbound_connectivity_check_path, 'w+') as outbound:
+                    outbound.write("Response code " + outbound_check + "\nWe found an issue with Outbound network connectivity from the cluster.")
+            return "Failed", storage_space_available
+
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An expection has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Outbound_Connectivity_Check_Fault_Type, summary="Error occured while performing outbound connectivity check in the cluster")
+            cli_outputs.append("An expection has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    except Exception as e:
+        logger.warning("An expection has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Outbound_Connectivity_Check_Fault_Type, summary="Error occured while performing outbound connectivity check in the cluster")
+        cli_outputs.append("An expection has occured while performing the outbound connectivity check on the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return "Incomplete", storage_space_available
+
+
+def check_msi_certificate(corev1_api_instance):
+
+    global cli_outputs
+
+    try:
+
+        # Initializing msi certificate as not present
+        msi_cert_present = False
+
+        # Going thorugh all the secrets in azure-arc
+        all_secrets_azurearc = corev1_api_instance.list_namespaced_secret(namespace="azure-arc")
+
+        for secrets in all_secrets_azurearc.items:
+
+            # If name of secret is azure-identity-certificate then we stop there
+            if(secrets.metadata.name == "azure-identity-certificate"):
```

Comment:
let's add this secret name in constansts.py file and import it from there instead of hard coding